server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -455,6 +455,42 @@ public void testStringIncludeExclude() throws Exception {
                     assertEquals("val010", result.getBuckets().get(1).getKeyAsString());
                     assertEquals(1L, result.getBuckets().get(1).getDocCount());
                     assertTrue(AggregationInspectionHelper.hasValue((InternalTerms)result));
+
+                    aggregationBuilder = new TermsAggregationBuilder("_name").userValueTypeHint(ValueType.STRING)
+                        .executionHint(executionHint)
+                        .includeExclude(new IncludeExclude("val00.+", null, null,
+                            new String[]{"val001", "val002", "val003", "val004", "val005", "val006", "val007", "val008"}))
```

Comment:
I appreciate that you added tests to make sure that the precedence is preserved with the new possible configurations, but I think we should test this on `IncludeExcludeTests` as well.

Reply:
I've refactored/added tests for all possible combinations of include/exclude in `IncludeExcludeTests`.